### ISSUE TITLE
ntopng multi-arch build failure on master

### ISSUE THREAD

**Issue by th0ma7** [SynoCommunity/spksrc#4028]
### Setup
_Package Name:_ ntopng
_Package Version:_ 3.8.1 (master).  While at it also note that a new v4.0 just got out in march.

_Problem Description:_ While working on cmake mk framework PR #3998 I ended getting a `ntopng` build failure on multiple arches as part of github actions.  Problem is currently present on master `spksrc`.

_NAS Architecture:_ `x64`, `evansport`, `aarch64`, `armv7-6.1`, `armv7-1.2`, `88f6281`, `qoriq`, `ppc853x-5.2`

#### Build log
```
/home/spksrc/git-master/spksrc/toolchains/syno-x64-6.1/work/x86_64-pc-linux-gnu/bin/x86_64-pc-linux-gnu-g++  src/CommunityIdFlowHash.o  src/Mac.o  src/AutonomousSystemHash.o  src/PeriodicActivities.o  src/ParserInterface.o  src/Trace.o  src/HostHash.o  src/LocalTrafficStats.o  src/PacketDumper.o  src/FlowHash.o  src/EthStats.o  src/FlowAlertCounter.o  src/VlanHash.o  src/GenericHash.o  src/Country.o  src/DummyInterface.o  src/TimeseriesRingStatus.o  src/PacketStats.o  src/IpAddress.o  src/AutonomousSystem.o  src/ViewInterface.o  src/FlowGrouper.o  src/Vlan.o  src/Host.o  src/VlanAddressTree.o  src/TimelineExtract.o  src/Flow.o  src/ExportInterface.o  src/ZCCollectorInterface.o  src/Mutex.o  src/MacManufacturers.o  src/Grouper.o  src/FrequentNumericItems.o  src/NetworkInterface.o  src/MDNS.o  src/ThreadPool.o  src/ThreadedActivity.o  src/CountriesHash.o  src/VirtualHostHash.o  src/NetworkStats.o  src/Redis.o  src/RemoteHost.o  src/StoreManager.o  src/LocalHost.o  src/AddressResolution.o  src/PacketDumperTuntap.o  src/Checkpointable.o  src/RwLock.o  src/ICMPstats.o  src/HTTPserver.o  src/HostPools.o  src/FrequentStringItems.o  src/MySQLDB.o  src/AddressList.o  src/Logstash.o  src/VirtualHost.o  src/Ntop.o  src/DB.o  src/TrafficStats.o  src/Paginator.o  src/DnsStats.o  src/NetworkDiscovery.o  src/nDPIStats.o  src/SNMP.o  src/TimeseriesExporter.o  src/Utils.o  src/LuaEngine.o  src/TimeseriesRing.o  src/GenericTrafficElement.o  src/GenericHashEntry.o  src/Prefs.o  src/StatsManager.o  src/AddressTree.o  src/HostTimeseriesPoint.o  src/AlertsManager.o  src/CollectorInterface.o  src/MacHash.o  src/NetworkInterfaceTsPoint.o  src/FrequentTrafficItems.o  src/main.o  src/TcpPacketStats.o  src/PcapInterface.o  src/TcpFlowStats.o  src/InterfaceStatsHash.o  src/NtopGlobals.o  src/ElasticSearch.o  src/HTTPstats.o  src/AlertCounter.o  src/Geolocation.o  src/ProtoStats.o  src/PF_RINGInterface.o  -Wall /home/spksrc/git-master/spksrc/spk/ntopng/work-x64-6.1/install/var/packages/ntopng/target/lib/libndpi.a -lpcap -L/home/spksrc/git-master/spksrc/spk/ntopng/work-x64-6.1/install/var/packages/ntopng/target/lib -lrrd   -L/home/spksrc/git-master/spksrc/spk/ntopng/work-x64-6.1/install/var/packages/ntopng/target/lib -lzmq -L/home/spksrc/git-master/spksrc/spk/ntopng/work-x64-6.1/install/var/packages/ntopng/target/lib -ljson-c -lmaxminddb -L/home/spksrc/git-master/spksrc/spk/ntopng/work-x64-6.1/install/var/packages/ntopng/target/lib -lsodium  -lsqlite3 -L/home/spksrc/git-master/spksrc/spk/ntopng/work-x64-6.1/install/var/packages/ntopng/target/lib -l   -L/home/spksrc/git-master/spksrc/spk/ntopng/work-x64-6.1/install/var/packages/ntopng/target/lib -lssl -lcurl -llua -lmysqlclient -lcrypto  -L/home/spksrc/git-master/spksrc/toolchains/syno-x64-6.1/work/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/sys-root/lib -L/home/spksrc/git-master/spksrc/spk/ntopng/work-x64-6.1/install//var/packages/ntopng/target/lib -Wl,--rpath-link,/home/spksrc/git-master/spksrc/spk/ntopng/work-x64-6.1/install//var/packages/ntopng/target/lib -Wl,--rpath,/var/packages/ntopng/target/lib  -L/usr/local/lib -lcap -lrt -lz -ldl -lcurl   -lm -lpthread -o ntopng
/home/spksrc/git-master/spksrc/toolchains/syno-x64-6.1/work/x86_64-pc-linux-gnu/bin/../lib/gcc/x86_64-pc-linux-gnu/4.9.3/../../../../x86_64-pc-linux-gnu/bin/ld: cannot find -l-L/home/spksrc/git-master/spksrc/spk/ntopng/work-x64-6.1/install/var/packages/ntopng/target/lib
collect2: error: ld returned 1 exit status
Makefile:162: recipe for target 'ntopng' failed
make[3]: *** [ntopng] Error 1
make[3]: Leaving directory '/home/spksrc/git-master/spksrc/spk/ntopng/work-x64-6.1/ntopng-3.8.1'
../../mk/spksrc.compile.mk:36: recipe for target 'compile_target' failed
make[2]: *** [compile_target] Error 2
make[2]: Leaving directory '/home/spksrc/git-master/spksrc/cross/ntopng'
../../mk/spksrc.depend.mk:44: recipe for target 'depend_target' failed
make[1]: *** [depend_target] Error 2
make[1]: Leaving directory '/home/spksrc/git-master/spksrc/spk/ntopng'
../../mk/spksrc.spk.mk:459: recipe for target 'arch-x64' failed
make: [arch-x64] Error 2 (ignored)
```

**Comment by hgy59** [comment]
This seems to be an issue with path length.
Your workpath is `/home/spksrc/git-master/spksrc/` that gives the error above.
I see the same error for the workpath `/github/workspace/` used in the github build actions.

With the default workpath in the container `/spksrc/` there is no such error.

**Comment by th0ma7** [comment]
That is soooo freaking strange?!?

**Comment by th0ma7** [comment]
Found it... was quite subtile but `ntopng` autobuild produced `Makefile` ended-up having the following:
```
NLIBS = $(NDPI_LIB) $(LIBPCAP) $(LIBRRDTOOL_LIB) $(LIBNBPF_LIB) $(ZEROMQ_LIB) $(JSON_LIB) -lmaxminddb $(SODIUM_LIB)  -lsqlite3 -L/spksrc/spk/ntopng/work-apollolake-6.1/install/var/packages/ntopng/target/lib -lssl -lcurl -llua -lmysqlclient -lcrypto  -L/spksrc/toolchains/syno-apollolake-6.1/work/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/sys-root/lib -L/spksrc/spk/ntopng/work-apollolake-6.1/install//var/packages/ntopng/target/lib -Wl,--rpath-link,/spksrc/spk/ntopng/work-apollolake-6.1/install//var/packages/ntopng/target/lib -Wl,--rpath,/var/packages/ntopng/target/lib  -L/usr/local/lib -lcap -lrt -lz -ldl -lcurl  $(ZSTD_LIB) -lm -lpthread
```
But on non-root `/spksrc` this variable would contain an empty `-l   -L`.  Turns out that this was fixed in the `spksrc/cross/ntopng/Makefile` with the following:
```
@$(RUN) sed -i 's| -L/spksrc/.*$(INSTALL_PREFIX)/lib -l   -L/spksrc/| -L/spksrc/|g' ./Makefile
```
which turns out only addressing the issue for root `/spksrc` that I've adjusted to simply getting rid of the extra `-l` such as:
```
@$(RUN) sed -i 's| -l   -L/| -L/|g' ./Makefile
```

**Comment by th0ma7** [comment]
closing, fixed with #4031
